tests/worker/topics/test_logging_topic.py

```diff
@@ -0,0 +1,134 @@
+from typing import cast
+
+import pytest
+from pytest_mock import MockerFixture
+
+from saturn_engine.core import TopicMessage
+from saturn_engine.core.error import ErrorMessageArgs
+from saturn_engine.utils.traceback_data import TracebackData
+from saturn_engine.worker.topics import LoggingTopic
+from saturn_engine.worker.topics.logger import Level
+
+
+def generate_exception(msg: str) -> BaseException:
+    return Exception(msg)
+
+
+@pytest.mark.asyncio
+async def test_logging_topic(mocker: MockerFixture) -> None:
+    messages = [
+        TopicMessage(id="0", args={"n": 1}),
+        TopicMessage(id="1", args={"n": 2}),
+    ]
+    error_messages = [
+        TopicMessage(
+            id="2",
+            args={
+                "error": ErrorMessageArgs(
+                    type="Exception",
+                    module="test_logging_topic.py",
+                    message="TEST_EXCEPTION_2",
+                    traceback=TracebackData.from_exception(
+                        generate_exception("TEST_EXCEPTION_2")
+                    ),
+                )
+            },
+        ),
+        TopicMessage(
+            id="3",
+            args={
+                "error": ErrorMessageArgs(
+                    type="Exception",
+                    module="test_logging_topic.py",
+                    message="TEST_EXCEPTION_3",
+                    traceback=TracebackData.from_exception(
+                        generate_exception("TEST_EXCEPTION_3")
+                    ),
+                )
+            },
+        ),
+    ]
+    ignorable_error_message = TopicMessage(
+        id="4",
+        args={
+            "error": ErrorMessageArgs(
+                type="Exception",
+                module="test_logging_topic.py",
+                message="TEST_EXCEPTION_4",
+                traceback=TracebackData.from_exception(
+                    generate_exception("TEST_EXCEPTION_4")
+                ),
+            )
+        },
+    )
+
+    log_patch = mocker.patch("logging.Logger.log", return_value=None)
```

Comment:
Don't mock, `pytest` has `caplog` for that (See `"tests/worker/services/test_logger.py"`)